tests/lambda_test.go

```diff
@@ -21,27 +22,42 @@ func TestLambda(t *testing.T) {
 	assert.Nil(t, err)
 	defer f.Close()
 	spec, _ := ioutil.ReadAll(f)
-	reqs := make(chan *http.Request, 1)
+	reqs := make(chan *http.Request, 100)
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		reqs <- r
 		w.WriteHeader(http.StatusOK)
 		_, _ = w.Write([]byte(`{"v":"R"}`))
 	}))
 	defer srv.Close()
 	os.Setenv("TEST_HOST", srv.URL)
+	port, err := getPort()
+	require.Nil(t, err)
+	os.Setenv("TEST_SERVER_PORT", port)
 
-	done := make(chan error)
+	done := make(chan error, 1)
 	rt, err := transportd.New(context.Background(), spec, serverfullgw.Lambda)
 	require.Nil(t, err)
-	rt.Exit = func() chan error {
-		return done
+	rt.Exit = done
+	go func() {
+		if runErr := rt.Run(); runErr != nil {
+			t.Log(runErr)
+		}
+	}()
+	stop := time.Now().Add(time.Second)
+	var resp *http.Response
+	var req *http.Request
+	for time.Now().Before(stop) {
```

Comment:
Question: what's the goal of this loop? It looks like this went from a test that made one request to one that makes multiple?

Reply:
The look exists after the first successful request but repeats with a backoff if the request fails. This helps account for variable startup times of the background server in different environments. The end result is that the tests will be less flakey.